test/test_model.py

```diff
@@ -3823,6 +3823,62 @@ def test_set_info(self, model: ops.Model, fake_script: FakeScript):
         with pytest.raises(TypeError):
             secret.set_info()  # no args provided
 
+    def test_set_content_then_info(self, model: ops.Model, fake_script: FakeScript):
+        fake_script.write('secret-set', """exit 0""")
+        fake_script.write('secret-get', """echo '{"foo": "bar"}'""")
+
+        secret = self.make_secret(model, id='q')
+        secret.set_content({'foo': 'bar'})
+        description = 'desc'
+        secret.set_info(description=description)
+
+        calls = fake_script.calls(clear=True)
+        assert calls == [
+            ['secret-set', f'secret://{model._backend.model_uuid}/q', mock.ANY],
+            ['secret-get', f'secret://{model._backend.model_uuid}/q', '--peek', '--format=json'],
+            [
+                'secret-set',
+                f'secret://{model._backend.model_uuid}/q',
+                '--description',
+                description,
+                mock.ANY,
+            ],
+        ]
+        # For this test we don't need to check that the content was in the temporary file, but we
+        # do want to be certain that there was a file passed to the call (but don't care about
+        # what the name of the file was).
+        assert calls[0][-1].startswith('foo#file=') and calls[0][-1].endswith('/foo')
+        assert calls[2][-1].startswith('foo#file=') and calls[2][-1].endswith('/foo')
+
+    def test_set_info_then_content(self, model: ops.Model, fake_script: FakeScript):
+        fake_script.write('secret-set', """exit 0""")
+
+        secret = self.make_secret(model, id='q')
+        description = 'desc'
+        secret.set_info(description=description)
+        secret.set_content({'foo': 'bar'})
+
+        calls = fake_script.calls(clear=True)
+        assert calls == [
+            [
+                'secret-set',
+                f'secret://{model._backend.model_uuid}/q',
+                '--description',
+                description,
+            ],
+            [
+                'secret-set',
+                f'secret://{model._backend.model_uuid}/q',
+                '--description',
+                description,
+                mock.ANY,
```

Comment:
I guess this represents secret content, but not obvious on the first read.

Wdyt about adding a comment, for example like so?

```suggestion
                mock.ANY,  # foo#file=<path with `bar` content>
```

Reply:
There are 8 other cases of `mock.ANY` in these tests, all, I believe, meaning "file with secret content". It seems odd to comment this one and not any of the others, particularly this in this test it does actually explain what it is by verifying it right after that line, which the others don't (although they generally validate the content in the file).

I'm not opposed to this, but it does seem like it should be done consistently, and it feels like that would be better in another PR (like the one that changed `ANY` to `mock.ANY`).

Reply:
I've reworked the tests to not use `mock.ANY`. It's a bit wordier and it's no longer consistent with the other tests, but it avoids all the issues.